src/genpy/generator.py

```diff
@@ -156,9 +156,17 @@ def default_value(msg_context, field_type, default_package):
                 return "b''"
         elif array_len is None: #var-length
             return '[]'
-        else: # fixed-length, fill values
+        else:
+            # fixed-length
             def_val = default_value(msg_context, base_type, default_package)
-            return '[' + def_val + '] * ' + str(array_len)
+            if base_type in [
+                'byte', 'int8', 'int16', 'int32', 'int64', 'uint16', 'uint32',
+                'uint64', 'float32', 'float64', 'string', 'bool'
```

Comment:
This code needs to handle all possible types. Therefore I think it needs to also check for `byte`.

It does not need to consider `char` and `uint8` though since those are already handled in [the if block above](https://github.com/ros/genpy/pull/74/files#diff-cda8e9d50594619a9ea61bdd3216b820R151).